### PR TITLE
Remove object component

### PR DESCRIPTION
**Pull request by lcdr**
Soooo, just a couple months after adding the object component, I'm removing it. Why? Well, it turns out it wasn't actually required! The object component was just a thin wrapper on top of a single `<a>` element. In the first iteration of the PR that added it, the component also had actual functionality, pulling object data from the API and populating the component with it, but that was factored out into a directive, with the component only handling the visual appearance. And since this is just an `<a>` element, we can simply apply the directive on that directly.

So, this PR does just that, replacing `<lux-object [luxFetch]="id">` with `<a [luxFetchObject]="id">`.

Note: Since the object component used the `routerLink` and `luxTooltip` directives, this makes things a bit more complicated. Angular apparently still doesn't have an easy way of programmatically applying directives. However manually instatiating them and calling their methods does work. Since `routerLink` is rather complex and provided by a dependency, I chose to just inherit from it, which takes care of all the calls automatically. `luxTooltip` is instead handled via composition and explicit calls, since we control its implementation.

### CODE REVIEW

File: src/app/gui/object/object.directive.ts [hunk 1]
```diff
@@ -1,30 +1,60 @@
...
-    this.objectComponent.title = `#${id}`;
+  @Input("luxFetchObject") set id(id: number) {
+    this.routerLink = "/objects/"+id;
+    this.element.nativeElement.textContent = `#${id}`;
     this.luJson.getObject(id).subscribe(this.onObject.bind(this));
   }
 
-  constructor(private luJson: LuJsonService, private objectComponent: ObjectComponent) { }
+  constructor(
+    private luJson: LuJsonService,
```
Comment (Xiphoseer) [inline src/app/gui/object/object.directive.ts:20]: Can you use `LuCoreDataService`?

File: src/app/missions/detail/detail.component.html [hunk 2]
```diff
@@ -162,31 +162,31 @@ <h4>Chat Bubbles</h4>
   <dl id="chat-bubbles">
     <div *ngIf="getChatState(textsLocale, 1); let text">
       <dt><blockquote class="chat-bubble" luxTooltip="chat_state_1" [innerHTML]="text"></blockquote></dt>
-      <dd>Spoken by <lux-object [luxFetch]="mission.offer_objectID"></lux-object> when approached before the mission is offered.</dd>
+      <dd>Spoken by <a [luxFetchObject]="mission.offer_objectID"></a> when approached before the mission is offered.</dd>
```
Comment (Xiphoseer) [inline src/app/missions/detail/detail.component.html:165]: If this is a standard anchor now, could that be written like so?
```suggestion
      <dd>Spoken by <a [luxFetchObject]="mission.offer_objectID"/> when approached before the mission is offered.</dd>
```

File: src/app/gui/object/object.directive.ts [hunk 3]
```diff
@@ -1,30 +1,60 @@
...
+import { ActivatedRoute, Router, RouterLinkWithHref } from "@angular/router";
 import { LuJsonService } from "../../services";
+import { TooltipDirective } from "../tooltip.directive";
 
 @Directive({
-  selector: "lux-object[luxFetch]"
+  selector: "a[luxFetchObject]"
 })
-export class ObjectDirective {
+export class ObjectDirective extends RouterLinkWithHref {
```
Comment (Xiphoseer) [inline src/app/gui/object/object.directive.ts:10]: Ah, I see. That's rather smart :D

**Review by Xiphoseer — APPROVED** [review]
Small nits

File: src/app/missions/detail/detail.component.html (hunk 2 above)
Comment (lcdr) [reply src/app/missions/detail/detail.component.html:165]: Angular doesn't seem to like it:
```
Error: app/missions/tasks/npc-task/npc-task.component.html:3:1 - error NG5002: Only void and foreign elements can be self closed "a"
3 <a [luxFetchObject]="task.target"/>.
  ~~
  ```

**Review by Xiphoseer — APPROVED** [review]

**Pull request merged**